### PR TITLE
feat: clear conversation content locally

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Solutions

- Clear content locally by deleting all the assets from the conversation and the messages 

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: persistence/src/commonMain/db_user/com/wire/kalium/persistence/Conversations.sq [hunk 1]
```diff
@@ -85,6 +85,13 @@ UPDATE Conversation
...
 
-
 updateAccess:
 UPDATE Conversation SET access_list= ?, access_role_list = ? WHERE qualified_id = ?;
+
+deleteAllMessages:
+DELETE FROM Message
+WHERE conversation_id IS :conversation_id;
+
+getMessagesByContentType:
```
Comment (yamilmedina) [inline persistence/src/commonMain/db_user/com/wire/kalium/persistence/Conversations.sq:95]: I think the name could be something like: `getAssetsMessagesByConversationId`. Only if we are not expected to pass a different content type in the query filter

**Review by yamilmedina — COMMENTED** [review]
A small suggestion about a particular query. But looking excellent so far, as always, Mr. 🥇

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt [hunk 2]
```diff
@@ -566,6 +575,21 @@ class ConversationDataSource(
         conversationDAO.deleteConversationByQualifiedID(idMapper.toDaoModel(conversationId))
     }
 
+    override suspend fun getConversationAssetMessages(
+        conversationId: ConversationId,
+    ): Either<CoreFailure, List<Message>> =
```
Comment (MohamadJaara) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt:580]:
```suggestion
    ): Either<StorageFailure, List<Message>> =
```

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt [hunk 3]
```diff
@@ -566,6 +575,21 @@ class ConversationDataSource(
...
+        wrapStorageRequest {
+            conversationDAO.getMessagesByContentType(
+                idMapper.toDaoModel(conversationId),
+                MessageEntity.ContentType.ASSET
+            ).map(messageMapper::fromEntityToMessage)
+        }
+
+    override suspend fun deleteAllMessages(conversationId: ConversationId): Either<CoreFailure, Unit> =
```
Comment (MohamadJaara) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt:588]:
```suggestion
    override suspend fun deleteAllMessages(conversationId: ConversationId): Either<StorageFailure, Unit> =
```

**Comment by trOnk12** [comment]
This PR diverges too much from the second PR too review this one so I am going to close this one in favor of : https://github.com/wireapp/kalium/pull/834,

**Pull request closed without merging**